### DISCUSSION TITLE
Testing Custom Kong Plugin

### DISCUSSION THREAD

**Discussion by narendrapatel** [Kong/kong#7690] [Help]
Hi,

Trying to migrate tests for a plugin initially developed on 0.13.0 to 2.5.0

For the same, built the container with source code from Kong git repo with tag = 2.5.0, but receiving the below error while running any plugin tests with busted.

```
Error → ...ns/eg-rate-limiting/tests/test_api_with_missing_rule.lua @ 6
eg-rate-limiting - missing rule
/kong/spec/helpers.lua:401: error loading plugin schemas: on plugin 'request-transformer': request-transformer plugin is enabled but not installed;
no plugin found; on plugin 'azure-functions': azure-functions plugin is enabled but not installed;
no plugin found; on plugin 'zipkin': zipkin plugin is enabled but not installed;
no plugin found; on plugin 'pre-function': pre-function plugin is enabled but not installed;
no plugin found; on plugin 'post-function': post-function plugin is enabled but not installed;
no plugin found; on plugin 'proxy-cache': proxy-cache plugin is enabled but not installed;
no plugin found; on plugin 'session': session plugin is enabled but not installed;
no plugin found; on plugin 'grpc-web': grpc-web plugin is enabled but not installed;
no plugin found

stack traceback:
	/kong/spec/helpers.lua:401: in function 'get_db_utils'
	...ns/eg-rate-limiting/tests/test_api_with_missing_rule.lua:10: in function <...ns/eg-rate-limiting/tests/test_api_with_missing_rule.lua:6>

ERROR: 1
```

I've tested Kong startup. It starts up correctly. Only while running tests via busted it gives above error.
Not sure why we have errors for other plugins.
Also for get_db_utils error highlighted above, have used it in the test code as:
```
local bp = helpers.get_db_utils("postgres")
```

Can someone please help or guide the right way to proceed with testing.

**Comment by narendrapatel** [comment]
Hi, Circling back to check if anyone has faced similar error?

**Comment by PaulFischer-Kong** [comment]
@gszr Would this relate to moving plugins to core.

**Reply by gszr** [reply]
It does not - these are all external plugins.

**Comment by gszr** [comment]
@narendrapatel,

It seems the plugins might not have been installed. Did you run `luarocks make` as part of your build process? Do you mind sharing your Dockerfile with us?
